docs/route.md

```diff
@@ -490,9 +490,13 @@ In example 2, since `Layout` is the top-level component in both routes, the DOM
 
 ---
 
-#### Authentication
+#### Redirection
 
-The RouterResolver's `onmatch` hook can be used to run logic before the top level component in a route is initializated. The example below shows how to implement a login wall that prevents users from seeing the `/secret` page unless they login.
+The RouterResolver's `onmatch` hook can be used to run logic before the top level component in a route is initializated. You can use either `m.route.set()` or `window.history.back()` `.forward()`, etc... When redirecting with the `history` API, the `onmatch` hook must return `new Promise(function(){})` to suspend the resolution of the current route. A fresh Promise must be returned every time since recycling a pending Promise leads to memory leaks. `m.route.set()` takes care of cancelling the resolution of the aborted route, there's no need to return a pending Promise in that case.
```

Comment:
First, a few minor nits:

1. `RouterResolver` &rarr; `RouteResolver`. (first sentence)
2. `initializated` &rarr; `initialized`. (first sentence)
3. You missed a comma after `window.history.back()`, and you only need a single period after "etc." in that list - it reads a little awkwardly with the ellipsis. (second sentence)

I know 1 and 2 were incorrect in the original docs, but we should probably correct them here while we're at it.

> A fresh Promise must be returned every time since recycling a pending Promise leads to memory leaks.

That sounds like a bug in Mithril to be honest. I'm okay with documenting that, but it just sounds odd behaviorally.

---

As for the rest, I feel this whole paragraph just needs recast. It's just worded super awkwardly, and just doesn't flow very well. Here's my idea of a possible fix, although I'm open for improvements (like if I missed some key detail):

```md
The RouteResolver's `onmatch` hook can be used to run logic before the top level component in a route is initialized, and potentially to suspend or change resolution of a route. You can use `m.route.set()`, `history.back()`, `history.forward()`, etc. within this method. When you redirect using the history API, you must return a promise from the `onmatch` hook to suspend resolution of the current route, and you must return a newly created one to avoid memory leaks. You don't need to return a pending promise when using `m.route.set()`, since it takes care of cancelling the resolution of the aborted route automatically.
```